libraries/browser-tracker-core/src/tracker/out_queue.ts

```diff
@@ -85,13 +58,20 @@ export function OutQueueManager(
   retryStatusCodes: number[],
   dontRetryStatusCodes: number[],
   idService?: string,
-  retryFailedRequests: boolean = true
+  retryFailedRequests: boolean = true,
+  onRequestSuccess?: (data: EventBatch) => void,
+  onRequestFailure?: (data: RequestFailure) => void
 ): OutQueue {
   type PostEvent = {
     evt: Record<string, unknown>;
     bytes: number;
   };
 
+  // A reference to the current batch of events being sent
+  // This is copied into the onRequestSuccess and onRequestFailure callbacks when called,
+  // and then cleared after the callbacks are run
+  let currentBatch: EventBatch = [];
```

Comment:
I don't really like that this is a "global" variable within the out queue manager. Could we instead just pass the current batch to the `setXhrCallbacks` function so that it can use it in the callbacks?

Reply:
This is now passed into setXhrCallbacks, and directly into `onreadystateupdate` for oversized payloads.